### PR TITLE
return eml_validate check in eml_otherEntity_to_dataTable

### PR DESCRIPTION
**Pull request by dmullen17**
I found that this function doesn't return an error if the `eml_validate` check fails.  Adding this in @jeanetteclark if you want to give it a quick review.  Curious to see if any of the unit tests fail now as well.

### CODE REVIEW

**Comment by jeanetteclark** [comment]
Hey Dominic I need to look closer at the second commit you made but probably won't have time this week. Did you test the case where you already have a bunch of dataTables, and want to convert a single otherEntity to a dataTable? I'm also not really sure what lines 718-721 are doing - seems like duplicate entities from the editor should be handled elsewhere if that is what it is doing

**Comment by dmullen17** [comment]
@jeanetteclark this what I mean by duplicate otherEntities created by the editor https://test.arcticdata.io/view/urn:uuid:d2710caa-9d77-4744-a76a-36a3e94705b7.  I didn't phrase it well and if you can improve on that definitely update the comment because that code isn't self-explanatory.  

Here's an example of the error I was getting.  It wasn't showing up before because when `validate_eml` was to `TRUE` then `eml_validate` would run, but it didn't actually cause an error, it just silently failed.  
```
cn_staging <- CNode('STAGING')
adc_test <- getMNode(cn_staging,'urn:node:mnTestARCTIC')
doc <- read_eml(getObject(adc_test, 'urn:uuid:d2710caa-9d77-4744-a76a-36a3e94705b7'))
# the first one has attributes and the second doesn't - remove it
doc$dataset$otherEntity[[2]] <- NULL
# Pass doc into `eml_otherEntity_to_dataTable`

# Here's the first line that causes problems 
otherEntity <- doc$dataset$otherEntity
# If you print otherEntity to the console it's a list of length 1 (because it used to a list of length 2 but setting the 2nd otherEntity to NULL doesn't update it to the correct format)

# Ends up getting compounded by this line 
otherEntity <- list(otherEntity)
# Now if you print it you'll see [[1]][[1]] notation 
```

File: R/eml.R [hunk 1]
```diff
@@ -712,17 +712,17 @@ eml_otherEntity_to_dataTable <- function(doc, index, validate_eml = TRUE) {
 
   ## set OE entityTypes to NULL and select the ones we want to use
 
-  if (length(eml_get_simple(doc$dataset$otherEntity, "entityName")) == 1){
+  if (length(eml_get_simple(doc$dataset$otherEntity, "entityName")) == 1) {
     ## prepare OE to copy
     otherEntity <- doc$dataset$otherEntity
+    ## If there are copies of entities from the editor then otherEnt can incorrectly be a list
```
Comment (jeanetteclark) [inline R/eml.R]:
```suggestion
    ## Handle case where otherEntity is in a list of length 1 (boxed)
```

Comment (jeanetteclark) [reply R/eml.R]: I think this comment is more clear

**Pull request merged**